Meadowlark-js/backends/meadowlark-postgresql-backend/src/repository/SqlHelper.ts

```diff
@@ -2,163 +2,269 @@
 // Licensed to the Ed-Fi Alliance under one or more agreements.
 // The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
 // See the LICENSE and NOTICES files in the project root for more information.
-import { DocumentUuid, MeadowlarkId } from '@edfi/meadowlark-core';
+import { ReferringDocumentInfo, DocumentUuid, MeadowlarkId } from '@edfi/meadowlark-core';
+import { PoolClient, QueryResult } from 'pg';
 import format from 'pg-format';
+import { MeadowlarkDocument, NoMeadowlarkDocument } from '../model/MeadowlarkDocument';
+import { MeadowlarkAlias } from '../model/MeadowlarkAlias';
 
-// SQL for Selects
+/**
+ * Executes a database begin transaction
+ * @param client database connector client.
+ */
+export async function beginTransaction(client: PoolClient) {
+  await client.query('BEGIN');
+}
+
+/**
+ * Executes a database rollback transaction
+ * @param client database connector client.
+ */
+export async function rollbackTransaction(client: PoolClient) {
+  await client.query('ROLLBACK');
+}
+
+/**
+ * Executes a database commit transaction
+ * @param client database connector client.
+ */
+export async function commitTransaction(client: PoolClient) {
+  await client.query('COMMIT');
+}
 
 /**
- * Returns the SQL query that does a reverse lookup to find the meadowlarkIds of documents that reference the given
- * meadowlarkIds.
+ * Returns a list of meadowlarkIds of documents that reference the given meadowlarkIds.
  *
- * @param referencedMeadowlarkIds the meadowlarkIds of the documents that are being referenced
- * @returns a SQL query to find the meadowlarkIds of the documents referencing the given meadowlarkIds
+ * @param client database connector client.
+ * @param referencedMeadowlarkIds the array of meadowlarkIds of the documents that are being referenced
+ * @returns an array of meadowlarkIds of the documents referencing the given meadowlarkIds
  */
-export function findReferencingMeadowlarkIdsSql(referencedMeadowlarkIds: MeadowlarkId[]): string {
-  return format(
+export async function findReferencingMeadowlarkIds(
+  client: PoolClient,
+  referencedMeadowlarkIds: MeadowlarkId[],
+): Promise<MeadowlarkId[]> {
+  const querySelect = format(
     `SELECT parent_meadowlark_id FROM meadowlark.references WHERE referenced_meadowlark_id IN (%L)`,
     referencedMeadowlarkIds,
   );
+  const queryResult: QueryResult<any> = await client.query(querySelect);
+  return ((queryResult?.rowCount ?? 0) > 0 ? queryResult.rows.map((ref) => ref.parent_meadowlark_id) : []) as MeadowlarkId[];
```

Comment:
Since `(queryResult?.rowCount ?? 0) > 0` is ugly, non-obvious, and repeated 11 times, let's extract it to a function like `function hasResults(queryResult: QueryResult<any>)`

Reply:
There are still a bunch of functions in `SqlHelper` that return SQL strings that are being used by Update and Upsert. 😮 The goal should be for `client.query()` to **never** be called outside of `SqlHelper`.